### PR TITLE
Change Makefile into CMake project

### PR DESCRIPTION
**Pull request by khanghua1505**
Hello! This change removes Makefiles and adds CMakeLists into keystone-sdk and lib.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+# Program and flags
+#-------------------------------------------------------------------------------
+
+include(macros.cmake)
+
+if (NOT CMAKE_BUILD_TYPE)
+    set(CMAKE_BUILD_TYPE    "Debug")
+else()
+    if ((NOT CMAKE_BUILD_TYPE STREQUAL "Debug") AND (NOT CMAKE_BUILD_TYPE STREQUAL "Release"))
+    message(FATAL_ERROR "CMAKE_BUILD_TYPE must either be Denug or Release instead of ${CMAKE_BUILD_TYPE}")
```
Comment (dkohlbre) [inline CMakeLists.txt:27]: typo: Denug -> Debug

**Comment by archshift** [comment]
Don't see any issues here, except obviously Travis failing to build.

@dkohlbre CMake is nice in a couple different ways. It allows users to create project files for their IDEs of choice, and use Makefile replacements such as Ninja if they choose. It also makes it incredibly easy to do out-of-tree builds, compared to Makefiles where things get much more challenging (figured that out myself trying to make the test builds happen out-of-tree).

File: CMakeLists.txt (hunk 1 above)
Comment (khanghua1505) [reply CMakeLists.txt]: Hi! Thanks for reviewing the my suggestion.

I fixed my problem in the newest commit and fixed typo error.
My reason for moving from make to Cmake that: CMake is very usefull for small or large project. CMake is cross-platform and mordern. CMake syntax is clear than Makefile. It's wonderful when you maintain and extend project.

**Comment by dkohlbre** [comment]
Thanks for fixing the travis!
We'll need to update keystone/dev to make this work there as well.

**Pull request merged**